cpp/src/copying/scatter.cu

```diff
@@ -34,50 +35,82 @@ using bit_mask::bit_mask_t;
 
 namespace cudf {
 namespace detail {
-  
-__global__ void invert_map(gdf_index_type gather_map[], const gdf_size_type destination_rows,
-                            gdf_index_type const scatter_map[], const gdf_size_type source_rows){
-  gdf_index_type source_row = threadIdx.x + blockIdx.x * blockDim.x;
-  if(source_row < source_rows){
-    gdf_index_type destination_row = scatter_map[source_row];
-    if(destination_row < destination_rows){
-      gather_map[destination_row] = source_row;
+
+
+template <typename value_type, typename InputIterator>
+struct scatter_to_gather {
+
+  scatter_to_gather(InputIterator first, InputIterator last, value_type default_value):
+    first(first), last(last), default_value(default_value) {}
+
+  __device__
+  value_type operator()(value_type in) {
+    InputIterator found = thrust::find(thrust::seq, first, last, in);
```

Comment:
Should this be `thrust::device` instead of `thrust::seq`? I imagine this is very slow as is

Reply:
No!  launching a thrust::device algorithm from device code requires Dynamic Parallelism (it's a kernel launch from a kernel).

In this case every thread is doing the find.

Reply:
Just got rid of this approach altogether. Materializing a gathermap was *much* faster than either of the above